diff -Nur binutils-2.42/bfd/hosts/x86-64linux.h binutils-2.42-new/bfd/hosts/x86-64linux.h
--- binutils-2.42/bfd/hosts/x86-64linux.h	2024-01-29 04:00:00.000000000 +0400
+++ binutils-2.42-new/bfd/hosts/x86-64linux.h	2024-12-08 10:27:17.101429989 +0400
@@ -29,6 +29,17 @@
 #include <sys/time.h>
 #include <sys/types.h>
 
+#ifdef __BIONIC__
+#define ELF_PRARGSZ (80)
+
+struct elf_siginfo
+{
+    int	si_signo;			/* signal number */
+    int	si_code;			/* extra code */
+    int	si_errno;			/* errno */
+};
+#endif
+
 /* We define here only the symbols differing from their 64-bit variant.  */
 #include <sys/procfs.h>
 #include <stdint.h>
diff -Nur binutils-2.42/bfd/trad-core.c binutils-2.42-new/bfd/trad-core.c
--- binutils-2.42/bfd/trad-core.c	2024-01-29 04:00:00.000000000 +0400
+++ binutils-2.42-new/bfd/trad-core.c	2024-12-08 10:27:17.101429989 +0400
@@ -38,6 +38,12 @@
 # define NBPG getpagesize()
 #endif
 
+#ifdef __BIONIC__
+#define UPAGES			1
+#define HOST_TEXT_START_ADDR	(u.start_code)
+#define HOST_STACK_END_ADDR	(u.start_stack + u.u_ssize * NBPG)
+#endif
+
 struct trad_core_struct
 {
   asection *data_section;
diff -Nur binutils-2.42/libiberty/getpwd.c binutils-2.42-new/libiberty/getpwd.c
--- binutils-2.42/libiberty/getpwd.c	2024-01-29 04:00:00.000000000 +0400
+++ binutils-2.42-new/libiberty/getpwd.c	2024-12-08 10:29:55.957429928 +0400
@@ -39,6 +39,10 @@
 #include <limits.h>
 #endif
 
+#ifdef __BIONIC__
+#include <linux/limits.h>
+#endif
+
 #include "libiberty.h"
 
 /* Virtually every UN*X system now in common use (except for pre-4.3-tahoe
